subsys/net/ip/Kconfig

```diff
@@ -304,6 +304,15 @@ config NET_UDP_CHECKSUM
 	  Enables UDP handler to check UDP checksum. If the checksum is invalid,
 	  then the packet is discarded.
 
+config NET_UDP_MISSING_CHECKSUM
+	bool "Accept missing checksum (IPv4 only)"
+	depends on NET_UDP && NET_IPV4
```

Comment:
How about:

NET_UDP_DROP_NO_CHECKSUM
or
NET_UDP_DROP_CHECKSUM_ZERO

And set them to false by default, so that by default incoming UDP with checksum zero (no checksum) is replied to?

Reply:
I am not sure we want to accept missing checksum by default. Since RFC says it's for debugging or higher protocol that doesn't care. In zephyr side, we don't allow any higher protocol to tweak UDP settings at runtime to disable the checksum computation. Which means whatever higher protocol is made in zephyr will anyway always have the checksum in UDP. So at runtime it's not going to happen to have a protocol that has checksum in TX, but none in RX.

Imo, unless we get the feature to tweak UDP settings at runtime (which is a totally different issues than what's being fixed here), let's not make it on by default.